rq/queue.py

```diff
@@ -754,8 +758,44 @@ def enqueue_many(self, job_datas: List['EnqueueData'], pipeline: Optional['Pipel
             List[Job]: A list of enqueued jobs
         """
         pipe = pipeline if pipeline is not None else self.connection.pipeline()
-        jobs = [
-            self._enqueue_job(
+        jobs_without_dependencies = []
+        jobs_with_dependencies_unmet = []
+        jobs_with_dependencies_met = []
+
+        job_datas_without_dependencies = [job_data for job_data in job_datas if not job_data.depends_on]
+        if job_datas_without_dependencies:
+            jobs_without_dependencies = [
+                self._enqueue_job(
+                    self.create_job(
+                        job_data.func,
+                        args=job_data.args,
+                        kwargs=job_data.kwargs,
+                        result_ttl=job_data.result_ttl,
+                        ttl=job_data.ttl,
+                        failure_ttl=job_data.failure_ttl,
+                        description=job_data.description,
+                        depends_on=job_data.depends_on,
+                        job_id=job_data.job_id,
+                        meta=job_data.meta,
+                        status=JobStatus.QUEUED,
+                        timeout=job_data.timeout,
+                        retry=job_data.retry,
+                        on_success=job_data.on_success,
+                        on_failure=job_data.on_failure,
```

Comment:
Since kwargs are mostly the same, mind putting these kwargs into a `kwargs` dict that can be passed on to both `create_job` calls in this method? All the kwargs should be the same except for `status`, right?

Reply:
Let me know if my solution looks correct